src/app/add-course/add-course.component.ts

```diff
@@ -17,14 +18,17 @@ export enum MODE {
   styleUrls: ['./add-course.component.less'],
 })
 export class AddCourseComponent implements OnInit {
+
   courseItem!: CourseItem;
 
-  constructor(private router: Router, public courseService: CourseServiceService, private route: ActivatedRoute) {}
+  constructor(private router: Router, public courseService: CourseServiceService, 
+    private route: ActivatedRoute,private spinner: NgxSpinnerService) {}
 
   courseId: string | undefined;
   pageMode: MODE = MODE.NONE;
 
   ngOnInit() {
+    this.spinner.show();
```

Comment:
Why we are showing and hiding the spinner in one method with a few milliseconds difference?

Reply:
Well to satisfy needs :) and show that i t works